### PR TITLE
feat: onboarding survey changes

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [X] I ran `npm run re:build`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Others/OnboardingSurveyModalUtils.res [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  (),
+)
+
+let constructOnboardingSurveyBody = values => {
+  open LogicUtils
+  let dictFromJson = values->getDictFromJsonObject
+  let hyperswitchValue = dictFromJson->getDictfromDict("hyperswitch")
+  let filterOutOtherKeys =
+    hyperswitchValue->Dict.keysToArray->Array.filter(value => value->String.includes("otherstring"))
+  let _ = filterOutOtherKeys->Array.map(otherKey => hyperswitchValue->Dict.delete(otherKey))
```
Comment (PritishBudhiraja) [inline src/screens/Others/OnboardingSurveyModalUtils.res:165]: use forEach

File: src/entryPoints/HyperSwitchApp.res [hunk 2]
```diff
@@ -86,6 +86,30 @@ let make = () => {
...
+        )}?keys=OnboardingSurvey`
+      let res = await fetchDetails(url)
+      let firstValueFromArray = res->getArrayFromJson([])->getValueFromArray(0, JSON.Encode.null)
+      let onboardingDetailsFilled =
+        firstValueFromArray->getDictFromJsonObject->getDictfromDict("OnboardingSurvey")
+      if onboardingDetailsFilled->Dict.keysToArray->Array.length === 0 {
+        setCompanyNameModal(_ => true)
+      } else {
+        setCompanyNameModal(_ => false)
+      }
```
Comment (PritishBudhiraja) [inline src/entryPoints/HyperSwitchApp.res:105]:
```
let val = onboardingDetailsFilled->Dict.keysToArray->Array.length === 0
setCompanyNameModal(_ => val)
```

File: src/screens/Others/OnboardingSurveyModalUtils.res (hunk 1 above)
Comment (Riddhiagrawal001) [reply src/screens/Others/OnboardingSurveyModalUtils.res]: Adressed

File: src/entryPoints/HyperSwitchApp.res (hunk 2 above)
Comment (Riddhiagrawal001) [reply src/entryPoints/HyperSwitchApp.res]: Adressed

**Review by PritishBudhiraja — DISMISSED** [review]

File: src/screens/Others/CompanyNameModal.res [hunk 3]
```diff
@@ -1,19 +1,86 @@
...
+    let valueField = ReactFinalForm.useField(field_name).input
+    let textField = ReactFinalForm.useField(`${field_name}_otherstring`).input
+
+    let textInput: ReactFinalForm.fieldRenderPropsInput = {
+      name: `${field_name}_otherstring`,
+      onBlur: _ev => {
+        let textFieldValue = textField.value->LogicUtils.getStringFromJson("")
+        let valueFieldValue =
+          valueField.value->LogicUtils.getArrayFromJson([])->LogicUtils.getStrArrayFromJsonArray
+        if textFieldValue->String.length > 0 {
```
Comment (JeevaRamu0104) [inline src/screens/Others/CompanyNameModal.res:13]: use utils function

File: src/screens/Others/CompanyNameModal.res [hunk 4]
```diff
@@ -1,19 +1,86 @@
...
+      },
+      onFocus: _ev => (),
+      value: textField.value,
+      checked: false,
+    }
+
+    <div className="flex gap-2 items-center">
+      <CheckBoxIcon
+        key={`${field_name}_otherstring`}
+        isSelected={textField.value->LogicUtils.getStringFromJson("")->String.length > 0}
```
Comment (JeevaRamu0104) [inline src/screens/Others/CompanyNameModal.res:31]: same here

**Review by PritishBudhiraja — DISMISSED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**